scipy/cluster/hierarchy.py

```diff
@@ -2471,8 +2471,11 @@ def fcluster(Z, t, criterion='inconsistent', depth=2, R=None, monocrit=None):
     Z : ndarray
         The hierarchical clustering encoded with the matrix returned
         by the `linkage` function.
-    t : float
-        The threshold to apply when forming flat clusters.
+    t : float or int
```

Comment:
This could really be any numeric that can be safely cast to `int` and `float `, correct? A boolean would work here as well. I like your new comment below, but am unsure if changing to "float or int" is an improvement.

Reply:
its more of a semantic hint to further reinforce the point that it is referring to number of clusters, an integer.. Making it explicit, speaking in python terms!. Without that hint, people may supply 2.0, thinking they are expected a float.

in fact, it might even be better to rename to `t` to `t_or_num_clust`

Reply:
in other places referring to kmeans, scipy already uses `k_or_guess`

Reply:
the generic term to use for `float or int` is `scalar

Reply:
I take your point @raamana, I think scalar is the right way to go

Reply:
sure - what about the renaming the variable to `t_or_num_clust`?